rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/interpreter/SymbolTableStoragingTest.kt

```diff
@@ -347,4 +348,50 @@ class SymbolTableStoragingTest {
                 actual = variables["NUM"] ?: error("Not found NUM")
         )
     }
+
+    @Test
+    fun multiThreadTest() {
+
+        val testNames = listOf("execPgmAndEvaluateStorage", "initPgmByStorageAndEvaluateResult",
+                "execLRPgmAndEvaluateStorage", "execRTPgmTwiceAndPreserveValues", "initPreExistingVariablesPgmByStorageAndEvaluateResult",
+                "sameVariablesButDifferentACTGRP", "sameVariablesAndSameACTGRP")
+
+        val fixedThreadPool = 10
+        val repeatTests = 100
+
+        val executor = Executors.newFixedThreadPool(fixedThreadPool)
+        repeat(repeatTests) {
+            for (testName in testNames) {
+                var simbolTableStoragingTest = SymbolTableStoragingTest()
+                val workerThread: Runnable = WorkerThread(simbolTableStoragingTest, testName)
+                executor.execute(workerThread)
+            }
+        }
+        executor.shutdown()
+
+        println("Waiting for all thread to finish...")
+        while (!executor.isTerminated) {
```

Comment:
A very little change request. It's better to use: Executor.awaitTermination with a reasonable timeout (60 secs) in order to avoid potentially infinite loop in case that executor.isTerminated never return true

Reply:
It sounds good